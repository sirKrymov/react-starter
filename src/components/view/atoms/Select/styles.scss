@import '../../../../styles/configs';

.select-wrapper {
  min-height: inherit;
  box-shadow: none;
  height: 100%;

  &--error {
    .select__control {
      border: 1px solid var(--error) !important;
    }
  }

  &--disabled {
    .select__control--is-focused {
      border: 1px solid var(--borderPrimary) !important;
    }

    .select__menu {
      display: none !important;
    }
  }
}

.select {
  min-height: inherit;

  &__control {
    border-radius: $border-radius !important;
    transition: border $anim-time-fast !important;
    background: var(--bgPrimary) !important;
    min-height: inherit !important;
    box-shadow: none !important;
    border: 1px solid var(--borderPrimary) !important;
    height: 100%;

    &--is-focused {
      border: 1px solid var(--thirdy) !important;
    }
  }

  &__value-container {
    padding-left: 16px !important;
    min-height: inherit !important;
  }

  &__menu {
    border-radius: $border-radius !important;
    box-shadow: $box-shadow-light !important;
    animation: mount $anim-time;
    overflow: hidden !important;
  }

  &__menu-list {
    padding: 0 !important;
  }

  &__option {
    &:hover {
      background: var(--primaryDark) !important;
      cursor: pointer !important;
    }
  }
}
