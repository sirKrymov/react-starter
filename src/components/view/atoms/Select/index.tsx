// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import ReactSelect, { OptionsType, ValueType } from 'react-select';
import cn from 'classnames';

import { UiSize } from 'components/view/atoms/UiSize';

import './styles.scss';

import { SelectOptionType } from 'types/components.interface';

interface IProps {
  getOptionValue?(option: SelectOptionType): string;
  getOptionLabel?(option: SelectOptionType): string;
  placeholder?: string;
  className?: string;
  disabled?: boolean;
  onChange(option: ValueType<SelectOptionType>): void;
  height?: 'xs' | 'sm' | 'md' | 'lg' | 'full';
  width?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'full' | 'content';
  error?: boolean;
  value: ValueType<SelectOptionType>;
  list: OptionsType<SelectOptionType>;
  id: string;
}

export function Select({
  getOptionValue,
  getOptionLabel,
  placeholder,
  className,
  disabled,
  onChange,
  height = 'md',
  error,
  value,
  width,
  list,
  id
}: IProps): ReactElement {
  const selectWrapperClass = cn(
    {
      'select-wrapper': true,
      'select-wrapper--disabled': disabled,
      'select-wrapper--error': error
    },
    className
  );

  return (
    <UiSize width={width} height={height}>
      <ReactSelect
        classNamePrefix="select"
        getOptionValue={getOptionValue}
        getOptionLabel={getOptionLabel}
        isSearchable={!disabled}
        placeholder={placeholder}
        className={selectWrapperClass}
        onChange={onChange}
        options={list}
        value={value}
        id={id}
      />
    </UiSize>
  );
}
