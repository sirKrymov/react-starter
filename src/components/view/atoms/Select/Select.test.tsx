import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Select } from '../Select';

describe('<Select> atom component', () => {
  const handleChange = jest.fn();

  const checkList = [
    { label: 'Test 1', value: 1 },
    { label: 'Test 2', value: 2 }
  ];

  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(
      <Select
        onChange={handleChange}
        value={null}
        list={checkList}
        id="check-select"
      />
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
