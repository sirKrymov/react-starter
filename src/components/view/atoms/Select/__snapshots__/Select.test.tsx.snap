// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<UiSize
  height="md"
>
  <StateManager
    className="select-wrapper"
    classNamePrefix="select"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    id="check-select"
    isSearchable={true}
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    value={null}
  />
</UiSize>
`;
